alert.go

```diff
@@ -558,7 +571,10 @@ func (a *AlertNode) runAlert([]byte) error {
 			var highestPoint *models.BatchPoint
 
 			var currentLevel alert.Level
-			if state, ok := a.states[b.Group]; ok {
+			a.cardinalityMu.RLock()
```

Comment:
Since this is common and references a field on the alert node I think it would be best to wrap this into a small function.